process.cabal

```diff
@@ -56,19 +61,21 @@ library
         System.Process
         System.Process.CommunicationHandle
         System.Process.CommunicationHandle.Internal
+        System.Process.Environment
+        System.Process.Environment.OsString
```

Comment:
I don't understand why we have two modules which export the exact same thing. What's that about?

Reply:
They don't. One uses `OsString`, one doesn't.

See:

* https://hackage.haskell.org/package/filepath-1.5.4.0/docs/System-FilePath.html
* https://hackage.haskell.org/package/filepath-1.5.4.0/docs/System-OsPath.html

And:

* https://hackage.haskell.org/package/directory-1.3.9.0/docs/System-Directory.html
* https://hackage.haskell.org/package/directory-1.3.9.0/docs/System-Directory-OsPath.html

Reply:
Oh, OK, one of them re-exports stuff from `System.Environment`. Why?

Reply:
To have symmetric API. Otherwise you can only get those functions from process if you use OsString.

Reply:
I don't understand why one would want to get them from `process`. They're already in `base`.

Reply:
There is no hard technical reason.

I think re-export makes sense, unless it's already in Prelude.

Reply:
As hasufell mentioned, I did this to be consistent with the other libraries that offer `OsPath/OsString` versions of an existing `FilePath` API (see also [path](https://hackage.haskell.org/package/path)).

However, it's fair to say that the situation here is slightly different, since those libraries merely translated an API that already existed within the library, whereas here we are adding an API that is new for the library itself.

Personally, I am indifferent on including a `FilePath` version, as I don't plan to use it myself. Happy to go with consensus.

Reply:
It's not a key part of the PR and it's easier to add it later than it is to remove it, so let's be cautious and omit it for now.